components/camel-jaxb/src/main/java/org/apache/camel/converter/jaxb/FilteringXmlStreamWriter.java

```diff
@@ -36,13 +36,18 @@
     NonXmlCharFilterer nonXmlCharFilterer = new NonXmlCharFilterer();
 
     private XMLStreamWriter writer;
+    private String encoding;
 
     /**
      * @param writer
      *            target writer to wrap.
+     * @param encoding
+     *            the encoding to write in header
+     *
      */
-    public FilteringXmlStreamWriter(XMLStreamWriter writer) {
+    public FilteringXmlStreamWriter(XMLStreamWriter writer, String encoding) {
         this.writer = writer;
+        this.encoding = encoding != null ? encoding.toUpperCase() : null;
```

Comment:
Could you please explain why we would need to store it upper case please ?

Reply:
It is nicer for consumers of the produced XML. While lowercase is allowed by the standard and parsers should support it, there are parsers that break on lowercase. None I know of that break on uppercase.

Standard also uses uppercase.
https://www.iana.org/assignments/character-sets/character-sets.xhtml

This might break the users expectation though, but in this case it might be nice to help the user do what will cause them the least pain. What do you think?